functime/feature_extraction/tsfresh.py

```diff
@@ -356,8 +354,14 @@ def c3(x: TIME_SERIES_T, n_lags: int) -> FLOAT_EXPR:
     float | Expr
     """
     twice_lag = 2 * n_lags
+    if isinstance(x, pl.Series):
+        if twice_lag >= x.len():
+            return np.nan
+        else:
+            return (x * x.shift(n_lags) * x.shift(twice_lag)).sum() / (x.len() - twice_lag)
+    else:
     # Would potentially be faster if there is a pl.product_horizontal()
-    return (x * x.shift(n_lags) * x.shift(twice_lag)).sum() / (x.len() - twice_lag)
```

Comment:
For series input, there is no .mul(), only _ _ mul _ _. 

Using * works for both series and expression because both has _ _ mul_ _.

Reply:
As I want to avoid the user to face this message while using this function in eager mode:

`ZeroDivisionError: division by zero`

I need to perform `twice_lag >= x.len()` and therefore I need to distinguish eager and lazy.

Ofc, I can do this as well:

```
twice_lag = 2*n_lags
if isinstance(x, pl.Series):
    if twice_lag >= x.len():
        return np.nan
return ((x *(x.shift(n_lags))* (x.shift(twice_lag))).sum()) / ((x.len() - twice_lag))
```